mlflow/models/utils.py

```diff
@@ -1247,26 +1315,29 @@ def _enforce_pyspark_dataframe_schema(
     return new_pf_input.drop(*columns_to_drop)
 
 
-def _enforce_datatype(data: Any, dtype: DataType, required=True):
+def _enforce_datatype(data: Any, dtype: DataType, required=True, enforce_param=False):
     if not required and _is_none_or_nan(data):
         return None
 
     if not isinstance(dtype, DataType):
         raise MlflowException(f"Expected dtype to be DataType, got {type(dtype).__name__}")
     if not np.isscalar(data):
         raise MlflowException(f"Expected data to be scalar, got {type(data).__name__}")
-    # Reuse logic in _enforce_mlflow_datatype for type conversion
-    pd_series = pd.Series(data)
-    try:
-        pd_series = _enforce_mlflow_datatype("", pd_series, dtype)
-    except MlflowException:
-        raise MlflowException(
-            f"Failed to enforce schema of data `{data}` with dtype `{dtype.name}`"
-        )
-    return pd_series[0]
+    if enforce_param:
+        return _enforce_param_datatype(data, dtype)
+    else:
+        # Reuse logic in _enforce_mlflow_datatype for type conversion
+        pd_series = pd.Series(data)
+        try:
+            pd_series = _enforce_mlflow_datatype("", pd_series, dtype)
+        except MlflowException:
+            raise MlflowException(
+                f"Failed to enforce schema of data `{data}` with dtype `{dtype.name}`"
+            )
+        return pd_series[0]
```

Comment:
nit: the else is unnecessary here since we return in the `if`

```suggestion
    # Reuse logic in _enforce_mlflow_datatype for type conversion
    pd_series = pd.Series(data)
    try:
        pd_series = _enforce_mlflow_datatype("", pd_series, dtype)
    except MlflowException:
        raise MlflowException(
            f"Failed to enforce schema of data `{data}` with dtype `{dtype.name}`"
        )
    return pd_series[0]
```